custom/icds_reports/models/manager.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import absolute_import, unicode_literals
+
+import uuid
+
+from django.db.models.manager import BaseManager
+from django.db.models.query import QuerySet
+
+from corehq.toggles import ICDS_COMPARE_QUERIES_AGAINST_CITUS, NAMESPACE_OTHER
+
+
+class CitusComparisonQuerySet(QuerySet):
+    def _fetch_all(self):
+        from custom.icds_reports.tasks import run_citus_experiment_raw_sql
+        if ICDS_COMPARE_QUERIES_AGAINST_CITUS.enabled(uuid.uuid4().hex, NAMESPACE_OTHER):
+            run_citus_experiment_raw_sql.delay(str(self.query), data_source=self.model.__name__)
```

Comment:
rather use `self.query.sql_with_params()` to get SQL and params separately

Reply:
Yes, this will not work if there are strings or dates in the parameter set, for example.

Reply:
Yeah. I asked about that in #celery a couple hours ago. for the moment I'm going to use `DjangoJSONEncoder` to handle this https://github.com/dimagi/commcare-hq/pull/24098/commits/0fc7a7962579cef554b30028e56923e2d05903c1#diff-2fc21c16c8147f60bf13421c5090b762R18 & https://github.com/dimagi/commcare-hq/pull/24098/commits/b26c7b50a6c5d6b075f8edfd7da82d3fd583e3ed#diff-515ee24d8bc1a09c528fcbf4764d4757R24

In a separate PR I'll add a new serializer that uses django by default http://docs.celeryproject.org/en/latest/userguide/calling.html#serializers This should help the move to json be much less of a headache